src/java.base/share/classes/sun/security/pkcs12/PKCS12KeyStore.java

```diff
@@ -1479,25 +1481,32 @@ private byte[] calculateMac(char[] passwd, byte[] data)
         final PBEParameterSpec params;
         final String algName;
         final MacData macData;
-        final String kdfHmac;
+        String kdfHmac;
         int writeIterationCount = macIterationCount;
 
-        if (macAlgorithm.startsWith("PBEWith") ||
-                defaultMacAlgorithm().startsWith("PBEWith")) {
-            if (defaultMacAlgorithm().equals("PBEWithHmacSHA512")) {
-                kdfHmac = "HmacSHA512";
-            } else if (defaultMacAlgorithm().equals("PBEWithHmacSHA256")) {
-                kdfHmac = "HmacSHA256";
+        if (newKeystore) {
```

Comment:
I cannot see how `newKeystore` is useful. Back when it's set to true, `macAlgorithm` was also set to `defaultMacAlgorithm()`. Therefore there is no need to try `defaultMacAlgorithm()` again below. As for `writeIterationCount`, for a new keystore it is -1, so we can use this to decide whether it should be assigned `defaultMacIterationCount()`.